src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4691,47 +4700,40 @@ private static bool IsParameterModifier(SyntaxToken token)
                 case SyntaxKind.ParamsKeyword:
                 // For error tolerance
                 case SyntaxKind.ReadOnlyKeyword:
-                case SyntaxKind.IdentifierToken when token.ContextualKind == SyntaxKind.ScopedKeyword:
                     return true;
             }
 
             return false;
         }
 
-        private void ParseParameterModifiers(SyntaxListBuilder modifiers, bool isFunctionPointerParameter = false)
+        private void ParseParameterModifiers(SyntaxListBuilder modifiers, bool isFunctionPointerParameter)
         {
-            while (IsParameterModifier(this.CurrentToken))
+            bool tryScoped = true;
+
+            while (IsParameterModifierExcludingScoped(this.CurrentToken))
             {
-                if (this.CurrentToken.ContextualKind == SyntaxKind.ScopedKeyword)
+                if (this.CurrentToken.Kind is SyntaxKind.RefKeyword or SyntaxKind.OutKeyword or SyntaxKind.InKeyword)
```

Comment:
>) [](http://example.com/codeflow?start=116&length=1)

`or SyntaxKind.ReadOnlyKeyword` perhaps?

Reply:
> ```or SyntaxKind.ReadOnlyKeyword``` perhaps?

That is not needed for any valid scenario. And it is really questionable if that would add any value for an error scenario. Going as is for now.